### ISSUE TITLE
Multiple pdf format fail with generated figures

### ISSUE THREAD

**Issue by Nenuial** [quarto-dev/quarto-cli#8367]
### Bug description

When you have multiple pdf formats in a Quarto document that contains figures (generated by R for example), the process cleans up those figure after rendering the first PDF and the second PDF fails to render.

### Steps to reproduce

````md
---
title: "Test"
format: 
  pdf: default
  pdf+option: default
---

```{r}
cars |> 
  ggplot2::ggplot(ggplot2::aes(x = speed, y = dist)) +
  ggplot2::geom_line()
```
````

### Expected behavior

The figures should be kept until all formats are rendered.

### Actual behavior

The figures are deleted after the first PDF rendered.

### Your environment

- IDE: RStudio Version 2023.12.0+369 (2023.12.0+369)
- MacOS Sonoma 14.2.1

### Quarto check output

Quarto 1.4.545
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.545
      Path: /Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: v2023.12
      Chromium: 869685

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /Users/admin/Library/TinyTeX/bin/universal-darwin
      Version: 2023

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.11.4 (Conda)
      Path: /opt/homebrew/Caskroom/miniconda/base/bin/python
      Jupyter: 5.3.1
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.3.2
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library
      knitr: 1.45
      rmarkdown: 2.25

[✓] Checking Knitr engine render......OK

**Comment by cderv** [comment]
This is related to other issues linked to 
* #6518 

We need to track input / output and intermediates totally to name them correctly and avoid overwrite

**Comment by andrewheiss** [comment]
There's a temporary workaround to this if you keep the intermediate markdown files around (see https://github.com/quarto-dev/quarto-cli/issues/8373#issuecomment-1979245883). This will work with Quarto 1.5.29 (and maybe earlier?):

````qmd
---
title: "Test"
format: 
  pdf: default
  pdf+option: default
keep-md: true
---

```{r}
cars |> 
  ggplot2::ggplot(ggplot2::aes(x = speed, y = dist)) +
  ggplot2::geom_line()
```

````

**Comment by andrewheiss** [comment]
Though according to @cscheid it's likely an accident that it's working: https://github.com/quarto-dev/quarto-cli/issues/6705#issuecomment-1706786384

**Comment by Nenuial** [comment]
Thanks for this!
It will, as long as this _accident_ works, offer a workaround that allows me to avoid manual renders.

**Comment by cderv** [comment]
I don't think this is an accident: when `keep-md: true` is set, the supporting files are kept. It is on purpose because in .md file, there will be linked to resources

https://github.com/quarto-dev/quarto-cli/blob/1ec01151ad59930b6042abed34ee9f90fd1826ab/src/command/render/cleanup.ts#L48-L55

So this works in this case, but you'll get leftovers at the end. 

The "accident" is this example 
````
---
title: "Test"
format: 
  pdf: default
  pdf+option:
    keep-md: true
---

```{r}
cars |> 
  ggplot2::ggplot(ggplot2::aes(x = speed, y = dist)) +
  ggplot2::geom_line()
```

````

Where compilation would work now, with no leftovers, but because we have an issue where `keep-md: true` is not really respected.

This will all be handled better with #6518

**Comment by Nenuial** [comment]
Ok, that good to know. Thanks for the explanation.

Looking forward to #6518 !!
